frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,50 @@
+import { Button } from '#libs/components/components.js';
+import { IconColor } from '#libs/enums/enums.js';
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useHandleClickOutside, useRef } from '#libs/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  children: React.ReactNode;
+  isDisplayed: boolean;
+  title: string;
+  onClose: () => void;
+};
+
+const Modal: React.FC<Properties> = ({
+  children,
+  isDisplayed,
+  title,
+  onClose,
+}) => {
+  const reference = useRef<HTMLDivElement>(null);
+
+  useHandleClickOutside({ reference, onClose });
```

Comment:
Check how we are handling references at now [here](frontend/src/libs/components/modal/modal.tsx) as @v0ldemar01 suggested. The same approach should be used in your modal wrapper

Reply:
Thank you! 
Will check in a minute